src/schema/v2/previewSavedSearch/suggestedFilters.ts

```diff
@@ -0,0 +1,198 @@
+import _ from "lodash"
+import artworkMediums from "lib/artworkMediums"
+import attributionClasses from "lib/attributionClasses"
+import { extractNodes, isExisty } from "lib/helpers"
+import { SearchCriteriaLabel } from "./searchCriteriaLabel"
+import { GraphQLFieldResolver } from "graphql"
+import { ResolverContext } from "types/graphql"
+import gql from "lib/gql"
+
+const ALLOWED_RARITY_SUGGESTIONS = ["unique", "limited edition"]
+const MAX_SUGGESTIONS = 2
+const MAX_ARTIST_SERIES_SUGGESTIONS = 5
+
+export const suggestedFilters: GraphQLFieldResolver<
+  any,
+  ResolverContext,
+  { [argName: string]: any }
+> = async (parent, args, context) => {
+  const { artistIDs } = parent
+  const { source } = args
+  const { filterArtworksLoader, gravityGraphQLLoader } = context
+
+  if (!artistIDs) {
+    throw new Error("artistIDs are required to get suggested filters")
+  }
+
+  if (source && (!source.type || !source.id)) {
+    throw new Error("type and id are both required for alert source")
+  }
+
+  const gravityArgs = {
+    size: 0,
+    aggregations: ["attribution_class", "medium", "artist_series"],
+    artist_ids: artistIDs,
+  }
+
+  const { aggregations } = await filterArtworksLoader(gravityArgs)
+
+  const suggestedFilters: SearchCriteriaLabel[] = []
+
+  let artistSeriesOptions: SearchCriteriaLabel[] | null
+
+  if (source?.type === "Artwork") {
+    artistSeriesOptions = await getArtistSeriesSearchCriteriaLabelsFromArtwork(
+      source,
+      gravityGraphQLLoader
+    )
+  } else {
+    artistSeriesOptions = getArtistSeriesSearchCriteriaLabelsFromArtist(
+      aggregations["artist_series"],
+      MAX_SUGGESTIONS
+    )
+  }
```

Comment:
**question**: Just to make sure I'm understanding this correctly — when creating an alert from an artwork source we return up to 5 artist series suggestions and when creating an alert from an artist source we return up to 2 artist series suggestion. Is that right?

Also, something that we've discussed but we should document/align on is what precedence these suggestions have over the existing medium+rarity suggestions. I think you made a good point recently that when creating an alert from an artwork source, the artist series suggestions are more relevant as they relate directly to the source artwork and not general inventory. So I might recommend something like this in terms of precedence and limit:

- Artwork source: `<artist series| max 5>, <medium | max 2>, <rarity | max 2>`
- Artist source: `<artist series | max >, <medium | max 2>, <rarity | max 2>`

Reply:
Yeah, these numbers are certainly debatable, but good idea to clarify the reasoning here:

- Artist source: up to 2 artist series suggestions, in order to be consistent with the existing pattern for medium and rarity

- Artwork source: up to 5 artist series suggestions, based on (a) the assumption that these will be highly relevant to the work in question, and (b) analysis of existing artist series inventory stats (see [here](https://github.com/artsy/force/pull/13244#discussion_r1419289315) — tldr  95% of works in artist series belong to no more than 2 series;  99.98% belong to no more than 5 series)

And yes, I should have noted that I _prepend_ the artist series suggestions to the list of all suggestions in this PR, based on that same assumption that these will be highly relevant to the artwork in question (though perhaps less so when based on overall artist aggregations.)

Reply:
Thanks! I started [a thread here in Slack](https://artsy.slack.com/archives/C05EQL4R5N0/p1702492997351529) to continue aligning on this.